datadogclient/datadog_client_test.go

```diff
@@ -175,623 +153,237 @@ var _ = Describe("DatadogClient", func() {
-				if metric.Tags[1] == "job:doppler" {
-					dopplerFound = true
-					Expect(metric.Points).To(Equal([]metrics.Point{
-						metrics.Point{
-							Timestamp: 1,
-							Value:     5.0,
-						},
-					}))
-				} else if metric.Tags[1] == "job:gorouter" {
-					gorouterFound = true
-					Expect(metric.Points).To(Equal([]metrics.Point{
-						metrics.Point{
-							Timestamp: 2,
-							Value:     76.0,
-						},
-					}))
-				} else {
-					panic("Unknown tag found")
-				}
-			}
-		}
-		Expect(dopplerFound).To(BeTrue())
-		Expect(gorouterFound).To(BeTrue())
-		validateMetrics(payload, 2, 0)
-	})
-
-	It("posts CounterEvents in JSON format and empties map after post", func() {
-		c.ProcessMetric(&events.Envelope{
-			Origin:    proto.String("origin"),
-			Timestamp: proto.Int64(1000000000),
-			EventType: events.Envelope_CounterEvent.Enum(),
-			CounterEvent: &events.CounterEvent{
-				Name:  proto.String("counterName"),
-				Delta: proto.Uint64(1),
-				Total: proto.Uint64(5),
-			},
-		})
-
-		c.ProcessMetric(&events.Envelope{
-			Origin:    proto.String("origin"),
-			Timestamp: proto.Int64(2000000000),
-			EventType: events.Envelope_CounterEvent.Enum(),
-			CounterEvent: &events.CounterEvent{
-				Name:  proto.String("counterName"),
-				Delta: proto.Uint64(6),
-				Total: proto.Uint64(11),
-			},
-		})
-
-		err := c.PostMetrics()
-		Expect(err).ToNot(HaveOccurred())
-
-		Eventually(bodies).Should(HaveLen(1))
-
-		var payload datadogclient.Payload
-		err = json.Unmarshal(bodies[0], &payload)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(5))
-		counterNameFound := false
-		for _, metric := range payload.Series {
-			Expect(metric.Type).To(Equal("gauge"))
-
-			if metric.Metric == "datadog.nozzle.origin.counterName" {
-				counterNameFound = true
+			} else if metric.Tags[0] == "test_tag:2" {
+				tag2Found = true
 				Expect(metric.Points).To(Equal([]metrics.Point{
 					metrics.Point{
-						Timestamp: 1,
-						Value:     5.0,
+						Timestamp: 1000,
+						Value:     0.0,
+					},
+					metrics.Point{
+						Timestamp: 1000,
+						Value:     1.0,
+					},
+					metrics.Point{
+						Timestamp: 1000,
+						Value:     2.0,
+					},
+					metrics.Point{
+						Timestamp: 1000,
+						Value:     3.0,
 					},
 					metrics.Point{
-						Timestamp: 2,
-						Value:     11.0,
+						Timestamp: 1000,
+						Value:     4.0,
 					},
 				}))
 			}
 		}
-		Expect(counterNameFound).To(BeTrue())
-		validateMetrics(payload, 2, 0)
 
-		err = c.PostMetrics()
-		Expect(err).ToNot(HaveOccurred())
-
-		Eventually(bodies).Should(HaveLen(2))
-
-		err = json.Unmarshal(bodies[1], &payload)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(payload.Series).To(HaveLen(3))
-
-		validateMetrics(payload, 2, 5)
+		Expect(tag1Found).To(BeTrue())
+		Expect(tag2Found).To(BeTrue())
 	})
 
-	It("sends a value 1 for the slowConsumerAlert metric when consumer error is set", func() {
```

Comment:
I can't tell if you removed this test or not? But I think it's an important test of the functionality that I can't find.

Reply:
The `slowConsumerAlert` metric was moved so that the nozzle itself owns it (so as to be able to add it to the metrics map prior to sending the map to the `Client` for posting), so now https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogfirehosenozzle/datadog_firehose_nozzle_test.go#L160 is responsible for checking it - which it does here: https://github.com/DataDog/datadog-firehose-nozzle/blob/2660d835d1b7d5fd7e21ba094ea9b2fc8c214a28/datadogfirehosenozzle/datadog_firehose_nozzle_test.go#L190

I could definitely add another test that explicitly checks that the value gets set when `nozzle.AlertSlowConsumerError()` gets called - I didn't do it originally for fear of redundancy, but perhaps it would be worth doing a separate test specifically for it?

Reply:
Ah I see, okay let's leave it. I noticed that it was there, but I missed one of the tests.